scripts/publish.rs

```diff
@@ -88,6 +88,7 @@ const PUBLIC_CRATES: &[&str] = &[
     "cranelift-bforest",
     "cranelift-codegen-shared",
     "cranelift-codegen-meta",
+    "cranelift-egraph",
```

Comment:
Now you have cranelift-egraph in this list in two places. I think that would only be necessary if there's a cyclic dependency between crates, and I think Cargo would refuse to build that... right?

Reply:
Different list actually! There is the `PUBLIC_CRATES` list (the line here) and the `CRATES_TO_PUBLISH` list and they are different for reasons I don't fully grok, but the crate needs to be in both for the publish script to be happy.

Reply:
Ohhh, that makes sense.